java/vector/src/main/java/org/apache/arrow/vector/FixedSizeBinaryVector.java

```diff
@@ -0,0 +1,401 @@
+      set(index, value);
+    } else {
+      BitVectorHelper.setValidityBit(validityBuffer, index, 0);
+    }
+  }
+
+  public void setSafe(int index, int isSet, byte[] value) {
+    handleSafe(index);
+    set(index, isSet, value);
+  }
+
+  /**
+   * Set the element at the given index to the given value.
+   *
+   * @param index  position of element
+   * @param buffer ArrowBuf containing binary value.
+   */
+  public void set(int index, ArrowBuf buffer) {
+    assert index >= 0;
+    assert byteWidth <= buffer.capacity();
+    BitVectorHelper.setValidityBitToOne(validityBuffer, index);
+    valueBuffer.setBytes(index * byteWidth, buffer, 0, byteWidth);
+  }
+
+  /**
+   * Same as {@link #set(int, ArrowBuf)} except that it handles the
+   * case when index is greater than or equal to existing
+   * value capacity {@link #getValueCapacity()}.
+   *
+   * @param index  position of element
+   * @param buffer ArrowBuf containing binary value.
+   */
+  public void setSafe(int index, ArrowBuf buffer) {
+    handleSafe(index);
+    set(index, buffer);
+  }
+
+  /**
+   * Set the element at the given index to the given value.
+   *
+   * @param index  position of element
+   * @param buffer ArrowBuf containing binary value.
+   */
+  public void set(int index, int isSet, ArrowBuf buffer) {
+    if (isSet > 0) {
+      set(index, buffer);
+    } else {
+      BitVectorHelper.setValidityBit(validityBuffer, index, 0);
+    }
+  }
+
+  /**
+   * Same as {@link #set(int, ArrowBuf)} except that it handles the
+   * case when index is greater than or equal to existing
+   * value capacity {@link #getValueCapacity()}.
+   *
+   * @param index  position of element
+   * @param buffer ArrowBuf containing binary value.
+   */
+  public void setSafe(int index, int isSet, ArrowBuf buffer) {
+    handleSafe(index);
+    set(index, isSet, buffer);
+  }
+
+  /**
+   * Set the variable length element at the specified index to the data
+   * buffer supplied in the holder
+   *
+   * @param index   position of the element to set
+   * @param holder  holder that carries data buffer.
+   */
+  public void set(int index, FixedSizeBinaryHolder holder) {
+    assert holder.byteWidth == byteWidth;
+    set(index, holder.buffer);
+  }
+
+  /**
+   * Same as {@link #set(int, FixedSizeBinaryHolder)} except that it handles the
+   * case where index and length of new element are beyond the existing
+   * capacity of the vector.
+   *
+   * @param index   position of the element to set
+   * @param holder  holder that carries data buffer.
+   */
+  public void setSafe(int index, FixedSizeBinaryHolder holder) {
+    handleSafe(index);
+    set(index, holder);
+  }
+
+  /**
+   * Set the variable length element at the specified index to the data
+   * buffer supplied in the holder
+   *
+   * @param index   position of the element to set
+   * @param holder  holder that carries data buffer.
+   */
+  public void set(int index, NullableFixedSizeBinaryHolder holder) {
```

Comment:
Should we call this `FixedSizeBinaryHolder` instead?

Reply:
There are both *Holders and Nullable*Holders.

Reply:
I see